src/javascript/SelectorTypes/SystemName/SystemName.jsx

```diff
@@ -11,6 +11,7 @@ import classes from './SystemName.scss';
 export const SystemNameCmp = ({field, value, values, id, editorContext, onChange}) => {
     const {t} = useTranslation();
 
+    const cleanedTitle = limitSystemNameIfNecessary(replaceSpecialCharacters(values['jcr:title'])?.toLowerCase(), field);
```

Comment:
This will be executed all the time adding extra calculation something when it's not necessary, best practice would be to put this into a function to be able to call it when necessary only.
from the isDisabled and the onClick.

Or at least use the same condition as: values['jcr:title'] !== undefined && editorContext.mode === Constants.routes.baseEditRoute